README.md

```diff
@@ -22,12 +26,89 @@ $ head dependencies.nix
 
 You can then use this to download all the necessary dependencies to run your application.
 
+### Building a Maven repository
+
+Now that you have a **nix** dependencies file; we can re-construct a Maven repository using Nix!
+
+```nix
+let pkgs = import <nixpkgs> {}
+    buildMavenRepsitory = import (
+        fetchTarball https://github.com/fzakaria/mvn2nix/archive/master.tar.gz
+        ).buildMavenRepsitory;
+in
+buildMavenRepsitory {}
+```
+
+This creates a **/nix/store** path which is a Maven repository that can be used, such as in `mvn package --offline -Dmaven.repo.local=${mavenRepository}`
+
+```bash
+$ tree /nix/store/2ps43297g5nii2k15kfy8z46fam51d8x-buildMavenRepsitory | head
+
+/nix/store/2ps43297g5nii2k15kfy8z46fam51d8x-buildMavenRepsitory
+├── com
+│   └── google
+│       ├── code
+│       │   └── findbugs
+│       │       └── jsr305
+│       │           └── 3.0.2
+│       │               └── jsr305-3.0.2.jar -> /nix/store/w20lb1dk730v77qis8l6sjqpljwkyql7-jsr305-3.0.2.jar
+│       ├── errorprone
+│       │   └── error_prone_annotations
+```
+
+### Sample Derivation
+
+```nix
+let pkgs = import <nixpkgs> {}
```

Comment:
It's generally a good idea to use default arguments instead, so that if you have multiple expression files you can still reuse a single application of nixpkgs when importing them.

```suggestion
{ pkgs ? import <nixpkgs> {} }:
let
```

`nix-build` etc will auto-call a top-level lambda, so no problem there.

Reply:
fixed!